bluebottle/auth/middleware.py

```diff
@@ -189,93 +189,68 @@ def process_request(self, request):
-
-        if authorized and self.logout_key and self.logout_key in request.GET:
-            if self.session_key in session:
-                del session[self.session_key]
-            url = request.path
-            querystring = request.GET.copy()
-            del querystring[self.logout_key]
-            if querystring:
-                url = '%s?%s' % (url, querystring.urlencode())
-            return self.redirect(request)
-
-        # Don't lock down if the user is already authorized for previewing.
-        if authorized:
-            return None
-
-        if form.is_valid():
-            if hasattr(form, 'generate_token'):
-                token = form.generate_token()
+            passwords = (common_settings.lockdown_password,)
+            form_data = request.method == 'POST' and request.POST or None
+            if self.form:
+                form_class = self.form
             else:
-                token = True
-            session[self.session_key] = token
-            return self.redirect(request)
+                form_class = get_lockdown_form(lockdown_settings.FORM)
+            form = form_class(passwords=passwords, data=form_data, **self.form_kwargs)
+            authorized = False
+            token = session.get(self.session_key)
+            if hasattr(form, 'authenticate'):
+                if form.authenticate(token):
+                    authorized = True
+            elif token is True:
+                authorized = True
+
+            if authorized and self.logout_key and self.logout_key in request.GET:
+                if self.session_key in session:
+                    del session[self.session_key]
+                url = request.path
+                querystring = request.GET.copy()
+                del querystring[self.logout_key]
+                if querystring:
+                    url = '%s?%s' % (url, querystring.urlencode())
+                return self.redirect(request)
+
+            # Don't lock down if the user is already authorized for previewing.
+            if authorized:
+                return None
 
-        page_data = {'until_date': until_date, 'after_date': after_date}
-        if not hasattr(form, 'show_form') or form.show_form():
-            page_data['form'] = form
+            if form.is_valid():
+                if hasattr(form, 'generate_token'):
+                    token = form.generate_token()
+                else:
+                    token = True
+                session[self.session_key] = token
+                return self.redirect(request)
 
-        response = render_to_response('lockdown/form.html', page_data)
-        response.status_code = 401
-        return response
+            page_data = {}
+            if not hasattr(form, 'show_form') or form.show_form():
+                page_data['form'] = form
+
+            response = render_to_response('lockdown/form.html', page_data)
+            response.status_code = 401
+            return response
+
+        return None
 
 
```

Comment:
Couldn't we greatly simplify this? No need to look at the session or the submitted form. We only need to check for the header being present right?